src/com/xilinx/rapidwright/util/RelocateDCPVertically.java

```diff
@@ -0,0 +1,392 @@
+/*
+ *
+ * Copyright (c) 2022 Xilinx, Inc.
+ * All rights reserved.
+ *
+ * Author: Pongstorn Maidee, Xilinx Research Labs.
+ *
+ * This file is part of RapidWright.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.xilinx.rapidwright.util;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.xilinx.rapidwright.design.ConstraintGroup;
+import com.xilinx.rapidwright.design.Design;
+import com.xilinx.rapidwright.design.Module;
+import com.xilinx.rapidwright.design.ModuleInst;
+import com.xilinx.rapidwright.design.Net;
+import com.xilinx.rapidwright.design.SiteInst;
+import com.xilinx.rapidwright.design.SitePinInst;
+import com.xilinx.rapidwright.device.PIP;
+import com.xilinx.rapidwright.device.Site;
+import com.xilinx.rapidwright.device.Tile;
+import com.xilinx.rapidwright.edif.EDIFHierNet;
+import com.xilinx.rapidwright.edif.EDIFNetlist;
+import com.xilinx.rapidwright.edif.EDIFPortInst;
+// Need util.Pair to allow compiling outside IDE with only dependent on RapidWright
+import com.xilinx.rapidwright.util.Pair;
+
+
+
+/**
+ * Fill some black boxes of a given design with a specific implementation.
+ */
+public class RelocateDCPVertically {
```

Comment:
Maybe move into `src.com.xilinx.design.tools` alongside `RelocationTools`?

Reply:
Rename to `RelocateAndInsertBlackboxes`?

Reply:
I originally use FillBlackBoxes, that describe the operation accurately. However, I change it because 
1) "FillBlackBoxes" seems applicable even for post synthesis design. 
2) We were calling the feature relocation. Thus, "Relocate" seems more appropriate and it indicates after post synthesis as well.

"Vertically" could be dropped from the name. I add it there so that users do not have a wrong expectation.

Reply:
How about `RelocateModulesIntoBlackboxes`?